resources/splunk_app.rb

```diff
@@ -69,25 +65,43 @@ def parse_meta_access(access)
     end
   end
 
+  # Provider exclusive methods
+
   action_class do
     include CernerSplunk::ProviderHelpers
 
+    def upgrade_app(strategy)
```

Comment:
The name of this method is kinda confusing because it's not really performing an upgrade with a particular strategy if that make sense. What about splitting it into `backup_app`  and `restore_app_parts(strategy)` ? 

Should we also have a `remove_backup` ? as it seems we could wind up with a backup already present and then what happens when we move the app to that place?